### ISSUE TITLE
Still random crashes for me.

### ISSUE THREAD

**Issue by Kiragroh** [bcatt09/DosimetryHelper#2]
Hi,

thanks for this nice GUI implementation. Do you still encounter crashes? My version always crashes after on plan build. At every second try, I get a crash without a helpful log entry

Thanks for reading

PS: I am using 16.1

**Comment by bcatt09** [comment]
Hi Max, is it causing Eclipse itself to crash or just the script? 

I'm using 16.1 also and was having issues with it causing Eclipse to crash sporadically and I was able to reproduce it by closing the import workflow window when a text field has keyboard focus.  The latest commit addressed this by clearing focus from the textboxes before closing the window, are you using this version?

If you're still having crashing issues with the latest version I may just need to not open additional windows for each module but instead keep them all within one window.  I've reached out to Varian about it a few times, but never gotten a response.  Looking at the Eclipse logs, the script itself appears to execute correctly and then something is causing Eclipse to crash during the saving process

I know it's not an ideal workflow, but for a temporary workaround, it doesn't seem to ever crash for me if I open one of the modules again after completing the import workflow and then just cancel it out before closing that main module selection window

**Comment by Kiragroh** [comment]
I implemented the RemoveFocus Lines in my adaption of your script and it still crashed. Mostly after the second time, I used a module.
But it works when using your newest script with updated dependencies. Maybe something I changed reversed your magic fix^^ 
I encountered these mysterious crashes with ESAPI before. Hopefully, the DosimetryHelper is fixed now. Resolving these issues is no fun. 
Thanks for this nice little GUI script. I had some scripts that have similar functionality, but your all-in-one solution makes more sense and it makes fun to read code from other ESAPI users.

Thank you for the last tip.

**Comment by bcatt09** [comment]
No problem, thanks for the interest!  Let me know if you find any other issues

**Comment by Kiragroh** [comment]
A little bit more info: I still got crashes when I closed the Import Workflow. I found the problem in the logs. There are some things from 'CanPerformUpdates' that cause the crashes. Maybe the RelayCommand is not stopping. For now, I commented some lines and all is fine now.

PS: your version will crash if someone opens it with a 2D image loaded. I would suggest changing a ine in 'CanGoToImportWorkflow' ->  return !double.IsNaN(_context.Image.Origin.x);

**Comment by bcatt09** [comment]
Oh, I did find some stuff in that section a few weeks ago and I think I made it more robust in one of my last commits.  I can't seem to get it to crash anymore, but maybe I'll just wrap that section in a try catch just to be safe.  

I think the crash with a 2D image loaded is happening because there's no structure set attached, so fixing that other open issue that I have (that I forgot about until you posted) fixes this also

Thanks for all your input!
